horace_core/sqw/@sqw/sqw.m

```diff
@@ -134,168 +132,172 @@
                 if isa(args.data_struct,'dnd_file_interface')
                     args.data_struct = obj.get_loader_struct_(...
                         args.data_struct,args.pixel_page_size);
+                    obj = from_class_struct(obj,args.data_struct);
+                elseif isfield(args.data_struct,'data')
+                    if isfield(args.data_struct.data,'version')
+                        obj = sqw.loadobj(args.data_struct);
+                    elseif isfield(args.data_struct.data,'serial_name')
+                        obj = serializable.from_struct(args.data_struct);
+                    else
+                        obj = from_class_struct(obj,args.data_struct);
+                    end
+                else
+                    error('HORACE:sqw:invalid_argument',...
+                        'Unidentified input data structure');
                 end
-                obj = obj.init_from_loader_struct_(args.data_struct);
+                
             end
-            
-            
         end
         
         % Public getters/setters expose all wrapped data attributes
         function val = get.data(obj)
             val = obj.data_;
         end
         function obj = set.data(obj, d)
-            %if isa(d,'data_sqw_dnd') || isempty(d)
-            obj.data_ = d;
-            %else
-            %    error('HORACE:sqw:invalid_argument',...
-            %        'Only data_sqw_dnd class or empty value may be used as data value. Trying to set up: %s',...
-            %        class(d))
-            %end
+            if isa(d,'data_sqw_dnd') || isempty(d)
+                obj.data_ = d;
+            else
+                error('HORACE:sqw:invalid_argument',...
+                    'Only data_sqw_dnd class or empty value may be used as data value. Trying to set up: %s',...
+                    class(d))
+            end
+        end
+        function hdr = get.header(obj)
+            %
+            if isempty(obj.experiment_info_)
+                hdr = IX_experiment();
+                return;
+            end
```

Comment:
there are huge number of unit tests, which talk to and check old header interface. I wrote hidden property which returns IX_experiment, having exactly this interface (may be alatt, angdeg should be added too)
this would save me a lot of typing and is convenient after all -- old header interface is useful in some places, so I do not see harm of exposing it. Inernaly it talks to Experiment so no violation for encapsulation. 
I think we should make header property public (read only as it is now)

Reply:
In that case what you have said in your reply should be put in a comment at the head of this function so it is clear what it is doing.

Reply:
well, I gave description around the property definition. (This is what hoes to help if property is not hidden, if hidden probably not.) I will copy something similar to getter.